src/measure_theory/integral/circle_transform.lean

```diff
@@ -174,4 +173,308 @@ begin
+    by {simp_rw [F, _root_.ae_strongly_measurable_iff_ae_measurable],
+    apply circle_transform_ae_measurable hR z x hx hf},
+  have hF_int : interval_integrable (F x) volume 0 (2 * π),
+    by {simp_rw F,
+      apply circle_interval_integrable hR z x hx hf},
+  have hF'_meas : ae_strongly_measurable (F' x) (volume.restrict (Ι 0 (2 * π))) ,
+    by {simp_rw [F', _root_.ae_strongly_measurable_iff_ae_measurable],
+    apply circle_transform_deriv_ae_measurable hR z x hx f hf},
+  have BOU := circle_transform_deriv_bound hR hx hf,
+  obtain ⟨bound, ε, hε ,h_ball, h_boun⟩ := BOU,
+  have h_bound : ∀ᵐ t ∂volume, t ∈ Ι 0 (2 * π) → ∀ y ∈ ball x ε , ∥F' y t∥ ≤ bound,
+    by {apply eventually_of_forall,
+    refine (λ _,(λ _, by {apply h_boun})) },
+  have bound_integrable : interval_integrable (λ _, bound) volume 0 (2 * π) ,
+    by {apply _root_.interval_integrable_const, },
+  have h_diff : ∀ᵐ t ∂volume, t ∈ Ι 0 (2 * π) → ∀ y ∈ ball x ε,
+  has_deriv_at (λ y, F y t) (F' y t) y,
+    by {simp_rw [F, F', circle_transform, circle_transform_deriv],
+    have := circle_transform_has_deriv_at R z f,
+    apply eventually_of_forall,
+    simp_rw [circle_transform, circle_transform_deriv] at this,
+    intros y hy x hx,
+    rw (interval_oc_of_le real.two_pi_pos.le) at hy,
+    have hy2 : y ∈ [0, 2*π], by {convert (Ioc_subset_Icc_self hy),
+      simp [interval_of_le real.two_pi_pos.le]},
+    apply this y x (h_ball hx),},
+  have := interval_integral.has_deriv_at_integral_of_dominated_loc_of_deriv_le hε hF_meas hF_int
+    hF'_meas h_bound bound_integrable h_diff,
+  simp only [F, has_deriv_at, has_deriv_at_filter, has_fderiv_within_at, mem_ball, zero_lt_mul_left,
+    inv_pos, zero_lt_bit0, zero_lt_one, norm_eq_abs,
+    interval_integral.interval_integrable_const] at *,
+  refine ⟨continuous_linear_map.smul_right (1 : ℂ →L[ℂ] ℂ)
+    (interval_integral (F' x) 0 (2 * π) volume) , _⟩,
+  apply (has_fderiv_at_filter.mono this.2 inf_le_left),
+end
+
+lemma circle_transform_sub (R : ℝ) (f g : ℂ → ℂ) (z w : ℂ) (θ : ℝ) :
+   ((circle_transform R z w f ) θ)-((circle_transform R z w g) θ) =
+  (circle_transform R z w (f - g) θ) :=
+begin
+  simp only [circle_transform, mul_inv_rev, inv_I, neg_mul, deriv_circle_map,
+    algebra.id.smul_eq_mul, neg_sub_neg, pi.sub_apply],
+  ring,
+end
+
+lemma circle_transform_sub_bound {R : ℝ} (hR : 0 < R) (f : ℂ → ℂ) (z w : ℂ) (r : ℝ)
+    (h : ∀ (x : sphere z R), (complex.abs (f x) ≤ abs r)) (θ : ℝ) :
+    complex.abs (circle_transform R z w f θ) ≤
+    complex.abs (circle_transform R z w (λ x, r) θ) :=
+begin
+  simp only [circle_transform, abs_of_real, mul_one, algebra.id.smul_eq_mul, abs_I, abs_two,
+  ←mul_assoc, deriv_circle_map, abs_circle_map_zero, mul_inv_rev, inv_I, absolute_value.map_neg,
+  absolute_value.map_mul, map_inv₀, one_mul],
+  apply_rules [monotone_mul_left_of_nonneg, mul_nonneg, mul_nonneg],
+  simp_rw inv_nonneg,
+  swap,
+  simp_rw inv_nonneg,
+  nlinarith,
```

Comment:
This looks weird - are you having several goals here? If yes, please use `{ .. }` to separate them.